remote-index.md

```diff
@@ -0,0 +1,150 @@
+support remote index flags in the [configuration file](/config.md) which will
+allow to point to different remote index servers depending on the project you
+are working on.
+
+## How it works
+
+Remote index uses the global static index built with `clangd-indexer` under
+the hood and replaces local static and background indices (see
+[indexing](/design/indexing.md) for more details on those). It contains all
+public symbols in the indexed project. The server (`clangd-index-server`
+instance) loads the index file and provides RPC service for the index
+requests (index-based code completion, references, etc). Clangd instance
+connects to it and forwards requests to index server whenever global index is
+needed. Unlike background index, remote index server does not rebuild the
+index incrementally and automatically - it just serves the given one, so it
+would most likely be more outdated. This problem is solved by manually
+rebuilding the whole static index (`clangd-indexer` can not update existing
+one incrementally yet) and replacing the file on disk: `clangd-index-server`
+watches index file updates and will use hot-reloading to ensure no-downtime
+index update.
+
+Our whole implementation is open source and you can browse the code at
+[clang-tools-extra/clangd/index/remote](https://github.com/llvm/llvm-project/tree/master/clang-tools-extra/clangd/index/remote).
+
+## Hosting the server
+
+If you would like to host the server for your project and share it with the
+developers, you should consider implementing infrastructure required to run
+the server and provide an up-to-date index version. Essentially, what you
+need providing a connection to `clangd-index-server` instance running
+somewhere and the supporting infrastructure to keep the index relatively
+up-to-date automatically. Developers working on your project would connect
+to the remote index server using `--remote-index-address` flag in their clangd
+invocation.
+
+The first thing you would need is the indexing job. Here's what it looks
+like:
+
+* Fetch the latest version of source code
+* Invoke build system tooling with the correct configuration to produce
+  generated files and the compilation database (`compile_commands.json`)
+* Run `clangd-indexer` to build static index
+
+And the data flow to assemble everything together might look like this:
+
+* Continuously check for newer source code version, if there is one --- run
+  indexing job, or simply do that periodically (every day/every N hours)
+* After the indexing job builds the index push it to the storage and tag it
+  with the latest version
+* On the server side, continuously check for new index versions in the storage.
+  As soon as the new index version is available, replace index file to allow
+  `clangd-index-server` hot-reloading the index.
+
+If you want to host remote indexing service for your project, please do not
+hesitate to reach out to us for assistance or any questions.
+
+### Remote index for LLVM
```

Comment:
rather than its own section, I would just mention this with a sentence after describing how server works. probably saying something like: "you can find the serving and indexing infrastructure for LLVM remote-index in xx"